components/api_server/src/initialization/migrations.py

```diff
@@ -120,6 +121,57 @@ def add_source_parameter_hash_to_latest_measurement(database: Database, report)
     return count
 
 
+def change_sonarqube_parameters(report) -> str:
+    """Replace the SonarQube parameters to adapt to the new (SonarQube 10.2) issue structure.
+
+    Return a description of the change, if any.
+    """
+    # Added after Quality-time v5.13.0, see https://github.com/ICTU/quality-time/issues/8354
+    changed = False
+    # Severity mapping conform https://docs.sonarsource.com/sonarqube/latest/user-guide/issues/#severity-mapping:
+    severity_mapping = {"blocker": "high", "critical": "high", "major": "medium", "minor": "low", "info": "low"}
+    for source in sources(
+        report,
+        metric_types=("security_warnings", "suppressed_violations", "violations"),
+        source_type="sonarqube",
+        parameter="severities",
+    ):
+        old_severities = source["parameters"]["severities"]
+        new_severities = {severity_mapping[severity] for severity in old_severities if severity in severity_mapping}
```

Comment:
Should we emit a warning message to log any unknown severities in the old param?

Reply:
I'm not sure we can do more than ignore the unknown parameter values, but I guess it can't hurt to log them.